### DISCUSSION TITLE
.Net 3.5 support

### DISCUSSION THREAD

**Discussion by ashishsinha24** [MarimerLLC/csla#3547] [Questions]
Hey,

I'm in a bit of a pickle. I have to assist an old client dealing with proprietary WinForms controls that only play nice with .NET 3.5, and to make things more interesting, they need to run on Windows 8. I'm keen on using the latest CSLA version, but I'm hitting a roadblock trying to reference the libraries.

Any ideas on how to tackle this?

Appreciate any suggestions!

Thanks

**Comment by rockfordlhotka** [comment]
What version of CSLA are you trying to reference? 

CSLA 7 still (I think) supports .NET 3.5. Maybe you have to go to CSLA 5. Either way, those versions support Windows 10.

**Reply by ashishsinha24** [reply]
Thanks

To ensure compatibility with third-party controls, the WinForms application must be developed using .NET Framework 3.5. Additionally, to maintain compatibility with Windows 8, other .NET Framework versions must not exceed 4.6.2.

Furthermore, the project requires a dashboard, which is planned to be implemented using Blazor.

**Reply by rockfordlhotka** [reply]
That is a challenge for sure. Blazor requires CSLA 7 or later. The good news is that even CSLA 8 will still target .NET 4.6.2.

Are you _sure_ the WinForms components won't work with .NET 4.6.2?

Regarding Windows 8 - the client should _really_ consider a way to update, as [Windows 8 is no longer under support](https://support.microsoft.com/en-us/windows/windows-8-1-support-ended-on-january-10-2023-3cfd4cde-f611-496a-8057-923fba401e93#Overview) and so they are increasingly vulnerable to security and other issues.
